apps/server/src/modules/synchronization/uc/synchronization.uc.ts

```diff
@@ -0,0 +1,106 @@
+import { Synchronization } from '../domain';
+import { SynchronizationStatusModel } from '../domain/types';
+import { SynchronizationLoggable } from '../domain/loggable';
+import { SynchronizationErrorLoggableException } from '../domain/loggable-exception';
+import { SynchronizationConfig } from '../config';
+
+@Injectable()
+export class SynchronizationUc {
+	constructor(
+		private readonly configService: ConfigService<SynchronizationConfig, true>,
+		private readonly schulconnexRestClient: SchulconnexRestClient,
+		private readonly synchronizationService: SynchronizationService,
+		private readonly userService: UserService,
+		private readonly logger: Logger
+	) {
+		this.logger.setContext(SynchronizationUc.name);
+	}
+
+	public async updateSystemUsersLastSyncedAt(systemId: string): Promise<void> {
+		this.logger.info(new SynchronizationLoggable('Start synchronization users from systemId', systemId));
+
+		const synchronizationId = await this.synchronizationService.createSynchronization();
+
+		try {
+			const usersToCheck = await this.findUsersToSynchronize(systemId);
+			const chunkSize = this.configService.get<number>('SYNCHRONIZATION_CHUNK');
+			const chunks = this.chunkArray(usersToCheck, chunkSize);
+			const promises = chunks.map((chunk) => this.updateLastSyncedAt(chunk, systemId));
+			const results = await Promise.all(promises);
+			const userSyncCount = results.reduce((acc, curr) => +acc + +curr, 0);
+
+			await this.updateSynchronization(synchronizationId, SynchronizationStatusModel.SUCCESS, userSyncCount);
+			this.logger.info(new SynchronizationLoggable('End synchronization users from systemId', systemId, userSyncCount));
+		} catch (error) {
+			if (error instanceof SynchronizationErrorLoggableException) {
+				const logMessage = error.getLogMessage();
+
+				await this.updateSynchronization(
+					synchronizationId,
+					SynchronizationStatusModel.FAILED,
+					0,
+					logMessage?.data?.errorMessage as string
+				);
+				this.logger.info(new SynchronizationLoggable('Failed synchronization users from systemId', systemId));
+			}
+		}
+	}
+
+	public async findUsersToSynchronize(systemId: string): Promise<string[]> {
+		let usersToCheck: string[] = [];
+		const usersDownloaded: SanisResponse[] = await this.schulconnexRestClient.getPersonenInfo({});
+
+		if (usersDownloaded.length === 0) {
+			throw new SynchronizationErrorLoggableException(`No users to check from systemId: ${systemId}`);
+		}
+		usersToCheck = usersDownloaded.map((user) => user.pid);
+
+		return usersToCheck;
+	}
+
+	public async updateLastSyncedAt(usersToCheck: string[], systemId: string): Promise<number> {
+		try {
+			const usersToSync = await this.userService.findByExternalIdsAndProvidedBySystemId(usersToCheck, systemId);
+
+			await this.userService.updateLastSyncedAt(usersToSync);
+
+			return usersToSync.length;
+		} catch {
+			throw new SynchronizationErrorLoggableException(`Problems with synchronization for systemId: ${systemId}`);
```

Comment:
Please reformat the message a little bit to make it more meaningful, e.g. "Failed to update lasySyncedAt field for users provisioned by system ${systemId}".

BTW you could also put the `systemId` in a separate field within the loggable, but I think it's not required if you're not adding it in every call, but occasionally for a few messages.

Reply:
done